### ISSUE TITLE
Datagrid: Issue with Keyword Search on a formatted (hyperlink) column  

### ISSUE THREAD

**Issue by ShyamaMenon** [infor-design/enterprise#3731]
1.	Go to https://master-enterprise.demo.design.infor.com/components/datagrid/example-keyword-search.html
2.	Click search and type “air” then hit enter
3.	Notice that the link looses its hyperlink styling

<img width="874" alt="Search" src="https://user-images.githubusercontent.com/41340433/78937808-19c50900-7a7f-11ea-82a9-420d8606dfa6.PNG">

Version

ids-enterprise: 4.26.1

**Comment by tmcconechy** [comment]
I think to fix this should skip certain formatted columns like button, alert, badge ect. For link it should try to highly the value in the link.

Or only check actual visual text somehow?

**Comment by hemantg05** [comment]
There is also issue with Keyword search making the Edit Formatter Button disappear and incorrectly showing it in search results.

![image](https://user-images.githubusercontent.com/1594815/81102044-51549300-8f0f-11ea-8877-055a7635993a.png)
![image](https://user-images.githubusercontent.com/1594815/81102060-56b1dd80-8f0f-11ea-8238-9583064291a5.png)

**Comment by jbrcna** [comment]
**QA Passed**
The issue is now resolved.
v4.29.0-dev
https://master-enterprise.demo.design.infor.com/components/datagrid/example-keyword-search.html
![image](https://user-images.githubusercontent.com/56722184/81930275-f1e73900-961a-11ea-9595-43b8d47e789a.png)

**Comment by hemantg05** [comment]
@jbrcna & @tmcconechy,  clicking on the highlighted link doesn't work when link text is searched. User should be able to navigate to the part specified by hyperlink. The example you provided doesn't have real hyperlinks to navigate to some other parts of the page or to another page. So, this use case can not be verified by that example.

If the non-link text is being searched, the hyperlinks are working. It's only when link text is searched that the hyperlinks don't work when clicked in the search result row.

cc @karinthulin

**Comment by tmcconechy** [comment]
The link is not set to do anything in this example @hemantg05 so i think its Ok. We dont really have anything to make these navigate to. But in a real case you can set the href or use the click event to navigate.

**Comment by hemantg05** [comment]
@tmcconechy, yes, we have set href in our application and it is not working. I just mentioned that such cases should also be covered during testing so that application developers don't have to find these issues and can be caught and fixed during development itself.

**Comment by tmcconechy** [comment]
I just tested this and it seems to work OK for me...
I changed the EP example to.. app/views/components/datagrid/example-index.html

```
columns.push({ id: 'productName', name: 'Product Name', field: 'productName', formatter: Formatters.Hyperlink, width: 300, minWidth: 100, href: 'http://www.google.com', target:'_blank'});
```        

Maybe something in your app router is going wrong. Either way if you can track down the issue lets make a seperate issue for this.

**Comment by tmcconechy** [comment]
I pushed that to the example for future testing https://github.com/infor-design/enterprise/commit/1e78ce577569c9d4ccf6c9ab1ea2318be42930f4

**Comment by hemantg05** [comment]
@tmcconechy, 

1. Regular DataGrid - no keyword search performed - WORKS
2. DataGrid - Keyword search performed - search text doesn't contain hyperlink text - WORKS
3. DataGrid - Keyword search performed - search text CONTAINS hyperlink text - NOT WORKING

Did you test the third scenario?

If there were was some issue with our app router, I think it wouldn't have worked for case 1 and case 2 as well. So, I suspect this is an IDS issue.

cc @karinthulin

**Comment by tmcconechy** [comment]
Yes i "think" i tested it but can you clarify steps? Example on master branch (not deployed yet but running so soon...) https://travis-ci.com/github/infor-design/enterprise/jobs/346621045

http://localhost:4000/components/datagrid/example-keyword-search.html

- link is google.com
- try any link it works
- search for o in the keyword search -> try any link it works
- clear search  -> try any link it works

What do you think @deep7102 ? See anything wrong here?

**Comment by hemantg05** [comment]
**Steps to reproduce:**

1. DataGrid with one column having hyperlinks (Can not use target:'_blank' because SohoDataGridColumn doesn't support such field. Gives typescript error)
2. Search exactly the Hyperlink text (e.g. If you see the hyperlink text as "Google" which points "https://www.google.com", then search "Google" in keyword search.
3. Click on the "Google" hyperlink which was returned as a search result.

**Expected Result**: App should navigate to the page/site/section specified by hyperlink
**Actual Result**: Nothing happens. Navigation isn't working.

cc @karinthulin

**Comment by tmcconechy** [comment]
OK i see. thanks for clarifying. i will reopen and @deep7102 can address that case

**Comment by janahintal** [comment]
The issue is now resolved. Thank you. 
Verified in 
https://master-enterprise.demo.design.infor.com/components/datagrid/example-keyword-search.html?theme=uplift&variant=light&colors=0066D4

![image](https://user-images.githubusercontent.com/45757016/84349749-917ff300-abea-11ea-8db3-44d47ff8ef03.png)
![image](https://user-images.githubusercontent.com/45757016/84349759-9644a700-abea-11ea-93f0-200ef1ce25af.png)

**Comment by dmolinaj** [comment]
Is there a way to add a different link per row?

**Comment by tmcconechy** [comment]
@dmolinaj you can use the href as a function or put `{{value}}` in the string that will be replaced with the value see https://github.com/infor-design/enterprise/blob/main/src/components/datagrid/datagrid.formatters.js#L261-L269

Also please use MS Teams for questions not posting on old closed issues
